### ISSUE TITLE
sometimes crashes when closed?

### ISSUE THREAD

**Issue by 32bitx64bit** [toasterofbread/spmp#185]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/toasterofbread/spmp/actions/workflows/android.yml).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce

Close app I believe?

### Expected behavior

Just. Closes 

### Actual behavior

Sometimes crashes

### Logs

```
type: crash
osVersion: google/redfin/redfin:14/UP1A.231005.007/2023101300:user/release-keys
package: com.toasterofbread.spmp:9
process: com.toasterofbread.spmp
processUptime: 306 + 77 ms
installer: com.android.packageinstaller

java.lang.RuntimeException: Unable to bind to service com.toasterofbread.spmp.platform.playerservice.PlatformPlayerService@d709853 with Intent { act=androidx.media3.session.MediaSessionService cmp=com.toasterofbread.spmp/.platform.playerservice.PlatformPlayerService }: java.lang.IllegalStateException
	at android.app.ActivityThread.handleBindService(ActivityThread.java:4713)
	at android.app.ActivityThread.-$$Nest$mhandleBindService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2279)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8199)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
Caused by: java.lang.IllegalStateException
	at androidx.core.os.BundleKt.checkStateNotNull(SourceFile:1)
	at androidx.media3.session.MediaSessionService.getServiceBinder(Unknown Source:5)
	at androidx.media3.session.MediaSessionService.onBind(SourceFile:110)
	at com.toasterofbread.spmp.platform.playerservice.PlatformPlayerService.onBind(Unknown Source:0)
	at android.app.ActivityThread.handleBindService(ActivityThread.java:4698)
	... 9 more

```

### SpMp version

576989f

### Android version

android 14

### Additional information

I have zero idea how this was caused, very much just assuming it was because I swipped to close it.

**Comment by 32bitx64bit** [comment]
Another crash when I attempt to download a song

```
type: ANR
osVersion: google/redfin/redfin:14/UP1A.231005.007/2023101300:user/release-keys
package: com.toasterofbread.spmp:9
process: com.toasterofbread.spmp
installer: com.android.packageinstaller

activity: com.toasterofbread.spmp/.MainActivity
cause: ANR Input dispatching timed out (95d4216 com.toasterofbread.spmp/com.toasterofbread.spmp.MainActivity (server) is not responding. Waited 5000ms for FocusEvent(hasFocus=false))
info: ANR in com.toasterofbread.spmp (com.toasterofbread.spmp/.MainActivity)
PID: 8340
Reason: Input dispatching timed out (95d4216 com.toasterofbread.spmp/com.toasterofbread.spmp.MainActivity (server) is not responding. Waited 5000ms for FocusEvent(hasFocus=false))
Parent: com.toasterofbread.spmp/.MainActivity
ErrorId: 0dc94cf0-7abf-439a-aa76-ee170297379f
Frozen: false
Load: 2.29 / 2.94 / 3.26
----- Output from /proc/pressure/memory -----
some avg10=0.14 avg60=0.13 avg300=0.03 total=158849074
full avg10=0.07 avg60=0.06 avg300=0.01 total=53821863
----- End output from /proc/pressure/memory -----
----- Output from /proc/pressure/cpu -----
some avg10=1.65 avg60=1.58 avg300=2.33 total=64151282952
----- End output from /proc/pressure/cpu -----
----- Output from /proc/pressure/io -----
some avg10=0.00 avg60=0.01 avg300=0.00 total=419083950
full avg10=0.00 avg60=0.00 avg300=0.00 total=310452092
----- End output from /proc/pressure/io -----

CPU usage from 2ms to 15457ms later (2023-10-27 18:56:45.036 to 2023-10-27 18:57:00.491):
  29% 2574/com.android.systemui: 23% user + 6.3% kernel / faults: 32188 minor 5 major
  18% 1658/system_server: 11% user + 6.3% kernel / faults: 21925 minor 52 major
  14% 648/surfaceflinger: 11% user + 3.2% kernel / faults: 17 minor 1 major
  7.6% 2986/com.android.launcher3: 6.2% user + 1.4% kernel / faults: 8606 minor 9 major
  5.9% 651/vendor.qti.hardware.display.composer-service: 3.4% user + 2.4% kernel
  4.4% 13291/com.android.bluetooth: 2.8% user + 1.6% kernel / faults: 11406 minor 378 major
  3.4% 1013/android.hardware.audio.service: 1.2% user + 2.2% kernel / faults: 2 minor
  2.5% 490/crtc_commit:105: 0% user + 2.5% kernel
  2.3% 491/crtc_event:105: 0% user + 2.3% kernel
  2.1% 1178/audioserver: 1.6% user + 0.4% kernel
15% TOTAL: 8.8% user + 5.6% kernel + 0.1% iowait + 0.7% irq + 0.1% softirq
CPU usage from 30ms to 856ms later (2023-10-27 18:56:45.065 to 2023-10-27 18:56:45.890):
  105% 1658/system_server: 47% user + 58% kernel / faults: 14725 minor 52 major
    53% 1722/Signal Catcher: 32% user + 21% kernel
    36% 8706/AnrAuxiliaryTas: 9.4% user + 26% kernel
    4% 8705/AnrConsumer: 0% user + 4% kernel
    1.3% 1658/system_server: 1.3% user + 0% kernel
    1.3% 1854/android.ui: 1.3% user + 0% kernel
    1.3% 1856/android.display: 1.3% user + 0% kernel
    1.3% 1882/ActivityManager: 0% user + 1.3% kernel
    1.3% 2482/VpnManagerServi: 0% user + 1.3% kernel
    1.3% 4690/binder:1658_19: 1.3% user + 0% kernel
    1.3% 4705/binder:1658_20: 0% user + 1.3% kernel
  17% 2574/com.android.systemui: 16% user + 1.3% kernel / faults: 345 minor
    10% 2574/ndroid.systemui: 9.5% user + 1.3% kernel
    2.7% 2617/RenderThread: 2.7% user + 0% kernel
    1.3% 2618/SysUiBg: 1.3% user + 0% kernel
  34% 15766/org.dslul.openboard.inputmethod.latin: 25% user + 9.4% kernel / faults: 2550 minor 20 major
    34% 15771/Signal Catcher: 21% user + 12% kernel
    3.1% 15766/putmethod.latin: 3.1% user + 0% kernel
  10% 1013/android.hardware.audio.service: 2.5% user + 7.7% kernel
    10% 3757/writer: 2.5% user + 7.7% kernel
  5.2% 1178/audioserver: 5.2% user + 0% kernel
    3.9% 2228/FastMixer: 2.6% user + 1.3% kernel
    1.3% 2229/AudioOut_15: 1.3% user + 0% kernel
  3.7% 648/surfaceflinger: 2.5% user + 1.2% kernel
    1.2% 648/surfaceflinger: 1.2% user + 0% kernel
  2.5% 651/vendor.qti.hardware.display.composer-service: 1.2% user + 1.2% kernel
    1.2% 748/HwBinder:651_2: 0% user + 1.2% kernel
  2.6% 1162/qrtr-ns: 1.3% user + 1.3% kernel
  1.2% 11/rcu_preempt: 0% user + 1.2% kernel
  1.2% 32/rcuop/2: 0% user + 1.2% kernel
  1.2% 40/rcuop/3: 0% user + 1.2% kernel
  1.2% 72/rcuop/7: 0% user + 1.2% kernel
  1.2% 321/kgsl_worker_thr: 0% user + 1.2% kernel
  1.2% 490/crtc_commit:105: 0% user + 1.2% kernel
  1.2% 491/crtc_event:105: 0% user + 1.2% kernel
  1.3% 1122/android.hardware.sensors@2.0-service.multihal: 1.3% user + 0% kernel / faults: 88 minor
    1.3% 28488/11101_see: 0% user + 1.3% kernel
  1.3% 1140/rlsservice: 0% user + 1.3% kernel
    1.3% 28493/rlsservice: 0% user + 1.3% kernel
  1.3% 1165/modem_svc: 0% user + 1.3% kernel
  1.3% 1206/tftp_server: 0% user + 1.3% kernel
  1.3% 1296/pm-service: 0% user + 1.3% kernel
  2.3% 4606/kworker/2:2-events: 0% user + 2.3% kernel
  2.4% 6992/kworker/u16:15-WCNSS_CNTL: 0% user + 2.4% kernel
  2.9% 13291/com.android.bluetooth: 2.9% user + 0% kernel
    2.9% 13322/HwBinder:13291_: 2.9% user + 0% kernel
  3.5% 17161/kworker/u16:6-CDSP_DATA: 0% user + 3.5% kernel
  3.5% 20797/kworker/u16:12-qmi_msg_handler: 0% user + 3.5% kernel
26% TOTAL: 12% user + 12% kernel + 0.9% iowait + 0.9% irq + 0.1% softirq

```

**Comment by toasterofbread** [comment]
Might be fixed by eec3c91ba7a32caf84bcf7e976e0d21b7766fb46, please try with that and report.

**Comment by 32bitx64bit** [comment]
Issue seems to be  resolved.
